syntax = "proto3";

option php_namespace= "App\\Proto";
option php_metadata_namespace = "App\\Proto\\GPBMetadata";

service Services{
    rpc InserirMatricula(Matricula) returns (InserirMatriculaRes) {}
    rpc AlterarNota(AlterarNotaReq) returns (MessageRes) {}
    rpc AlterarFaltas(AlterarFaltasReq) returns (MessageRes) {}
    rpc ListarDisciplinas(DisciplinasReq) returns (DisciplinasRes) {}
    rpc ListarAlunos(AlunosReq) returns (AlunosRes) {}
}


message ListaDisciplina {
    int32 RA = 1;
    string nome = 2;
    float nota = 3;
    int32 faltas = 4;
}


message DisciplinasReq {
    int32 ano = 1;
    int32 semestre = 2;
}

message DisciplinasRes {
    repeated ListaDisciplina disciplinas = 1;
    string message = 2;
}

message MessageRes {
    string message = 1;
}

message AlterarNotaReq {
    int32 RA = 1;
    string cod_disciplina = 2;
    float nota = 3;
}

message AlterarFaltasReq {
    int32 RA = 1;
    string cod_disciplina = 2;
    int32 faltas = 3;
}


message Matricula {
    int32 RA = 1;
    string cod_disciplina = 2;
    int32 ano = 3;
    int32 semestre = 4;
    float nota = 5;
    int32 faltas = 6;
}

message InserirMatriculaRes {
    Matricula matricula = 2;
    string message = 3;
}





message AlunosReq {
    string cod_disciplina = 1;
    int32 ano = 2;
    int32 semestre = 3;
}

message AlunosRes {
    repeated Aluno alunos = 1;
    string message = 2;
}


message Aluno {
    int32 RA = 1;
    string nome = 2;
    int32 periodo = 3;
    int32 cod_curso = 4;
}
